model_building/design_space.py

```diff
@@ -286,25 +287,25 @@ def generate_experiment_configurations(self, prefix, regression_inputs):
 
             if self._technique == ec.Technique.LR_RIDGE:
                 point = lr.LRRidgeExperimentConfiguration(self._campaign_configuration, hyperparams_point_values,
-                                                          regression_inputs, prefix)
+                                                          regression_inputs.copy(), prefix)
             elif self._technique == ec.Technique.XGBOOST:
                 point = xgb.XGBoostExperimentConfiguration(self._campaign_configuration, hyperparams_point_values,
-                                                           regression_inputs, prefix)
+                                                           regression_inputs.copy(), prefix)
             elif self._technique == ec.Technique.DT:
                 point = dt.DecisionTreeExperimentConfiguration(self._campaign_configuration, hyperparams_point_values,
-                                                               regression_inputs, prefix)
+                                                               regression_inputs.copy(), prefix)
             elif self._technique == ec.Technique.RF:
                 point = rf.RandomForestExperimentConfiguration(self._campaign_configuration, hyperparams_point_values,
-                                                               regression_inputs, prefix)
+                                                               regression_inputs.copy(), prefix)
             elif self._technique == ec.Technique.SVR:
                 point = svr.SVRExperimentConfiguration(self._campaign_configuration, hyperparams_point_values,
-                                                       regression_inputs, prefix)
+                                                       regression_inputs.copy(), prefix)
             elif self._technique == ec.Technique.NNLS:
                 point = nnls.NNLSExperimentConfiguration(self._campaign_configuration, hyperparams_point_values,
-                                                         regression_inputs, prefix)
+                                                         regression_inputs.copy(), prefix)
             elif self._technique == ec.Technique.STEPWISE:
                 point = sw.StepwiseExperimentConfiguration(self._campaign_configuration, hyperparams_point_values,
-                                                           regression_inputs, prefix)
+                                                           regression_inputs.copy(), prefix)
```

Comment:
For readability's sake, I'd rather have a declaration of another variable (or an update of `regression_inputs`?) which is then passed in all these if-clauses, instead of a copy being produced on the spot in multiple lines

Reply:
You're right! After some more investigation I've found out that the problem could also be solved in the experiment configuration class, as you can see from the next commit